src/test/java/com/rultor/agents/github/ClosePullRequestTest.java

```diff
@@ -0,0 +1,156 @@
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.rultor.agents.github;
+
+import com.jcabi.github.Comment;
+import com.jcabi.github.Issue;
+import com.jcabi.github.Repo;
+import com.jcabi.github.mock.MkGithub;
+import com.jcabi.xml.XMLDocument;
+import com.rultor.spi.Agent;
+import com.rultor.spi.Profile;
+import com.rultor.spi.Talk;
+import java.io.IOException;
+import org.apache.commons.lang3.StringUtils;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.Test;
+import org.xembly.Directives;
+
+/**
+ * Tests for {@link ClosePullRequest}.
+ *
+ * @author Viktor Kuchyn (kuchin.victor@gmail.com)
+ * @version $Id$
+ * @since 2.0
+ * @checkstyle ClassDataAbstractionCoupling (200 lines)
+ */
+public final class ClosePullRequestTest {
+
+    /**
+     * ClosePullRequest closes pull request if rebase mode.
+     * @throws Exception If error
+     */
+    @Test
+    public void closesPullRequestForRebaseMode() throws Exception {
+        final Repo repo = new MkGithub().randomRepo();
+        final Issue issue = repo.issues().create("", "");
+        final Profile profile = new Profile.Fixed(
+            new XMLDocument(
+                StringUtils.join(
+                    "<p><entry key='merge'>",
+                    "<entry key='rebase'>true</entry>",
+                    "</entry></p>"
+                )
+            )
+        );
+        final Agent agent = new ClosePullRequest(profile, repo.github());
+        final Talk talk = ClosePullRequestTest.createTalk(repo, issue);
+        agent.execute(talk);
+        MatcherAssert.assertThat(
+            new Issue.Smart(issue).state(), Matchers.is("closed")
+        );
+    }
+
+    /**
+     * ClosePullRequest leaves issue open if no rebase mode.
+     * @throws Exception If error
+     */
+    @Test
+    public void leavesPullRequestOpenWhenNoRebaseMode() throws Exception {
+        final Repo repo = new MkGithub().randomRepo();
+        final Issue issue = repo.issues().create("", "");
+        final Profile profile = new Profile.Fixed(
+            new XMLDocument(
+                StringUtils.join(
+                    "<p> <entry key='merge'>",
+                    "  <entry key='rebase'>false</entry>",
+                    "</entry> </p>"
+                )
+            )
+        );
+        final Agent agent = new ClosePullRequest(profile, repo.github());
+        final Talk talk = ClosePullRequestTest.createTalk(repo, issue);
+        agent.execute(talk);
+        MatcherAssert.assertThat(
+            new Issue.Smart(issue).state(), Matchers.is("open")
+        );
+    }
+
+    /**
+     * ClosePullRequest can comment issue with explanation.
+     * @throws Exception If error
+     */
+    @Test
+    public void commentsToIssueWithExplanation() throws Exception {
```

Comment:
@longtimeago yes. But after merging tests will break principle of single responsibility. After failing - not clear what exactly was broken - closing issue, or commenting. I'd prefer to leave 2 test cases